File: data/data.code.javascript.client/AckTestXX.js
module.exports = AckTestXX;

var Packet = require('../net/Packet');


function AckTestXX() {
	this._id_u8 = 0;
	this._id_u16 = 0;
	this._id_u32 = 0;
	this._repeat_id_u8 = new Array();
	this._optional_id_u8_flag = 0;
	this._optional_id_u8 = undefined;


	this.Decode = function(packet) {
		this._id_u8 = packet.ReadByte();
		this._id_u16 = packet.ReadUshort();
		this._id_u32 = packet.ReadUint();
		var repeat_id_u8_count = packet.ReadUshort();
		for (var i = 0; i < repeat_id_u8_count; i++)
		{
			this._repeat_id_u8.push(packet.ReadByte());
		}
		this._optional_id_u8_flag = packet.ReadByte();
		if (this._optional_id_u8_flag == 1)
		{
			this._optional_id_u8 = packet.ReadByte();
		}
	}
}

AckTestXX.prototype = {
	set IdU8(val) {
		this._id_u8 = val;
	},
	get IdU8() {
		return this._id_u8;
	},

	set IdU16(val) {
		this._id_u16 = val;
	},
	get IdU16() {
		return this._id_u16;
	},

	set IdU32(val) {
		this._id_u32 = val;
	},
	get IdU32() {
		return this._id_u32;
	},

	set RepeatIdU8(val) {
		this._repeat_id_u8 = val;
	},
	get RepeatIdU8() {
		return this._repeat_id_u8;
	},

	set OptionalIdU8(val) {
		this._optional_id_u8_flag = 1;
		this._optional_id_u8 = val;
	},
	get OptionalIdU8() {
		return this._optional_id_u8;
	},
}

File: examples/lua/client/protocols/AckTestXX.lua
module("AckTestXX", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckTestXX})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.id_u8 = 0
	self.id_u16 = 0
	self.id_u32 = 0
	self.repeat_id_u8 = {}
	self.optional_id_u8 = 0
end


function decode(self, pack)
	self.id_u8 = pack:readU8()
	self.id_u16 = pack:readU16()
	self.id_u32 = pack:readU32()
	repeat_id_u8_count = pack:readU16()
	for i = 1, repeat_id_u8_count do
		table.insert(self.repeat_id_u8, pack:readU8())
	end
	 local optional_id_u8_flag = pack:readU8()
	if optional_id_u8_flag then
		self.optional_id_u8 = pack:readU8()
	end
	return self
end
